Core/Src/main.c

```diff
@@ -87,16 +94,25 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
-  while (1)
-  {
+  while (1) {
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+	  HAL_SPI_TransmitReceive(&hspi1, txData, rxData, SIZE, TIMEOUT);
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+	  val = val | rxData[1];
+	  val = val << 14;
+	  val = val >> 8;
```

Comment:
So going 14 to the left then 8 to the right is equivalent to shifting 14-8=6 to the left... why not just bitshift 6 to the left but also why 6 and not 8 (1 byte)? If we don't shift by 8 then we are not making enough room for rxData[2].

Reply:
Oops, it should be 6, not 8. Bit shifting left by 14 is meant to ensure that all the preceding bits are 0.